include/integration/simple_MC_integration.hpp

```diff
@@ -0,0 +1,179 @@
+typedef typename HPolytope<Point>::MT MT;
+typedef typename HPolytope<Point>::VT VT;
+enum volumeType { CB , CG , SOB }; // Volume type for polytope
+VT Origin(0); // undefined 0 dimensional VT
+
+typedef const unsigned int Uint; // positive constant value for no of samples & dimensions
+
+// To return ||X||^2 for a VT
+NT normSquared(VT& X){
+    NT sum=0;
+    for( int i=0; i < X.rows() ; i++ ){
+        sum = sum + X(i)*X(i);
+    }
+    return sum;
+}
+
+// To check if two n-dimensional points ensure proper limits in integration 
+bool legitLimits(VT LL, VT UL){
+    if( UL.rows() == LL.rows() ) {
+        for ( int i = 0 ; i< LL.rows() ; i++){
+            if( LL(i) > UL(i) ){
+                std::cout << "Invalid integration limits\n";
+                return false;
+            }
+        }
+        return true;
+    }else{
+        std::cout << "Invalid integration limits\n";
+        return false;
+    }
+}
+
+// Hyper-Rectangle volume calculator in n-dimensions
+NT hyperRectVolume(VT LL, VT UL){
+    NT product=1;
+    if(legitLimits(LL,UL)){
+        for(int i=0; i<LL.rows(); ++i){
+            product = product * ( UL(i) - LL(i) );
+        }
+        return product;
+    }
+    else return 0;
+}
+
+// To sample a point between two n-dimensional points
+VT samplerBWLimits(VT LL, VT UL){
+    VT sample_point(LL.rows()); NT x;
+    for(int i=0; i<LL.rows(); ++i){
+            sample_point(i) = LL(i) + (NT)(rand()) / ((NT)(RAND_MAX/(UL(i) - LL(i))));     
+    }
+    return sample_point;
+}
+
+// Simple MC Integration over Hyper-Rectangles
+template < typename Functor >
+void SimpleMCIntegrate(Functor &Fx, Uint N ,VT LL, VT UL){
+    NT sum = 0;
+    if(legitLimits(LL,UL)){
+        for (int i = 1; i < N; i++ ) {
+            sum = sum + Fx(samplerBWLimits(LL,UL));
+        }    
+        std::cout << "Integral Value: " << hyperRectVolume(LL,UL) * sum / N << "\n";
+    }else{
+        std::cout << "Invalid integration limits\n";
+    }
+}
+
+template < typename Functor , typename Polytope >
+void SimpleMCPolytopeIntegrate(Functor &Fx, Polytope &P, Uint N,volumeType vType=CB,VT newOrigin=Origin){
+
+    // Polytope volumetric calculation params
+    Uint dim = P.dimension();
+    int walk_length = 10 + dim/10; 
+    NT e=0.1;
+    NT volume=0;
+
+    // Volume calculation for HPolytope
+    switch(vType)
+    {
+    case CB:     
+        volume = volume_cooling_balls<BallWalk, RNGType, HPOLYTOPE>(P, e, walk_length).second;
+        break;
+    case CG: 
+        volume = volume_cooling_gaussians<GaussianBallWalk, RNGType>(P, e, walk_length);
+        break;
+    case SOB: 
+        volume = volume_sequence_of_balls<BallWalk, RNGType>(P, e, walk_length);
+        break;    
+    default:  
+        volume = 0;
+        std::cout << "Error in enum:vType\n";
+        break;
+    }
+    
+    // For implementation of ReHMC walks
+    std::pair<Point, NT> inner_ball = P.ComputeInnerBall();
```

Comment:
the same for the random walk, that could be an input parameter of you integration function, just as in the volume functions https://github.com/GeomScale/volume_approximation/blob/develop/include/volume/volume_cooling_balls.hpp#L834
then you will not need all those HMC specific things and you could use you function with any walk that sample from the uniform distribution (since you assume uniformity of the sample)